test/objectid.test.js

```diff
@@ -126,5 +126,23 @@ describe('ObjectID', function() {
       const found = await Article.findOne({where: {title: 'abc'}});
       found.xid.should.be.an.instanceOf(ds.ObjectID);
     });
+
+    it('handles auto-generated PK properties defined in LB4 style', async () => {
+      const Note = ds.createModel('NoteLB4', {
+        id: {
+          type: 'string',
+          id: true,
+          generated: true,
+          mongodb: {dataType: 'ObjectID'},
```

Comment:
This is the property definition used in LB4.

When the primary keys is auto-generated by the database, juggler replaces the type provided by the user (`string`) with the type defined by the connector (`mongodb.ObjectID`).

As a result, we end up with a property defined as `{type: mongodb.ObjectID, mongodb: {dataType: 'ObjectID'}}`, which is something we were not handling well before my pull request.